### PR TITLE
AKCORE-186: Added Integration tests for Read and Write share group state RPCs

### PR DESCRIPTION
**Pull request by chirag-wadhwa5**
This PR contains integration tests for Read and Write share group state RPCs which read and write to the default persister.

Reference: [AKCORE-186](https://confluentinc.atlassian.net/jira/software/c/projects/AKCORE/issues/AKCORE-186?jql=project%20%3D%20%22AKCORE%22%20ORDER%20BY%20created%20DESC)

[AKCORE-186]: https://confluentinc.atlassian.net/browse/AKCORE-186?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: core/src/test/java/kafka/test/api/ShareConsumerTest.java [hunk 1]
```diff
@@ -1459,6 +1464,234 @@ public void testLsoMovementByRecordsDeletion() {
...
+
+        KafkaShareConsumer<byte[], byte[]> shareConsumer = createShareConsumer(new ByteArrayDeserializer(), new ByteArrayDeserializer(), "group1");
+        shareConsumer.subscribe(Collections.singleton(tp.topic()));
+
+        long duration = 45 * 1000; // 45 seconds
+
+        long startTime = System.currentTimeMillis();
+
+        // Poll for 45 seconds to ensure that the __share_group_state topic is created.
+        while (System.currentTimeMillis() - startTime < duration) {
```
Comment (smjn) [inline core/src/test/java/kafka/test/api/ShareConsumerTest.java:1483]: can we replace with `waitUntilTrue`

Comment (chirag-wadhwa5) [reply core/src/test/java/kafka/test/api/ShareConsumerTest.java]: Thanks for the review. Made the required changes in the latest commit.

File: core/src/test/java/kafka/test/api/ShareConsumerTest.java [hunk 2]
```diff
@@ -1459,6 +1464,243 @@ public void testLsoMovementByRecordsDeletion() {
...
+
+        long startTime = System.currentTimeMillis();
+
+        // Poll for 1 second to ensure acknowledgement is sent for the fetched record
+        while (System.currentTimeMillis() - startTime < duration) {
+            shareConsumer.poll(Duration.ofMillis(2000));
+        }
+
+        topics = listTopics().names().get();
+        assertTrue(topics.contains(SHARE_GROUP_STATE_TOPIC_NAME));
```
Comment (smjn) [inline core/src/test/java/kafka/test/api/ShareConsumerTest.java:1655]: use wait for condition

File: core/src/test/java/kafka/test/api/ShareConsumerTest.java [hunk 3]
```diff
@@ -1459,6 +1464,243 @@ public void testLsoMovementByRecordsDeletion() {
...
+
+        // Poll for 1 second to ensure acknowledgement is sent for the fetched record
+        while (System.currentTimeMillis() - startTime < duration) {
+            shareConsumer.poll(Duration.ofMillis(2000));
+        }
+
+        shareConsumer.close();
+
+        topics = listTopics().names().get();
+        assertTrue(topics.contains(SHARE_GROUP_STATE_TOPIC_NAME));
```
Comment (smjn) [inline core/src/test/java/kafka/test/api/ShareConsumerTest.java:1587]: waitForCondition

File: core/src/test/java/kafka/test/api/ShareConsumerTest.java [hunk 4]
```diff
@@ -1459,6 +1464,243 @@ public void testLsoMovementByRecordsDeletion() {
...
+            numRecordsConsumed += records.count();
+        }
+
+        assertEquals(1, numRecordsConsumed);
+
+        long duration = 1000; // 1 seconds
+        long startTime = System.currentTimeMillis();
+
+        // Poll for 1 second to ensure acknowledgement is sent for the fetched record
+        while (System.currentTimeMillis() - startTime < duration) {
```
Comment (smjn) [inline core/src/test/java/kafka/test/api/ShareConsumerTest.java:1546]: waitForCondition

**Review by smjn — CHANGES_REQUESTED** [review]
add waitForCondition checks when waiting for SGS topic

**Review by smjn — APPROVED** [review]
LGTM, thanks for the PR

File: core/src/test/java/kafka/test/api/ShareConsumerTest.java (hunk 2 above)
Comment (chirag-wadhwa5) [reply core/src/test/java/kafka/test/api/ShareConsumerTest.java:1655]: Thanks for the review. Actually at this point I ma not waiting for the creation of internal topic, that would have been created automatically before the consumer consumes its first message. the extra 1 second poll has been added to make sure any remaining acknowledgements for the consumed messages are sent to the broker.

File: core/src/test/java/kafka/test/api/ShareConsumerTest.java (hunk 3 above)
Comment (chirag-wadhwa5) [reply core/src/test/java/kafka/test/api/ShareConsumerTest.java:1587]: Same reason as mentioned above

File: core/src/test/java/kafka/test/api/ShareConsumerTest.java (hunk 4 above)
Comment (chirag-wadhwa5) [reply core/src/test/java/kafka/test/api/ShareConsumerTest.java:1546]: Same reason as mentioned above

**Pull request merged**